### PR TITLE
Various fixes and improvements

### PR DESCRIPTION
**Pull request by tklengyel**
Signed-off-by: Tamas Lengyel <tamas.lengyel@intel.com>

### CODE REVIEW

**Comment by schumilo** [comment]
Looks good! Thanks!

**Comment by schumilo** [comment]
FYI: I reverted one of your commits (f4b5a29d501fcd9bce0bec530707b8ab8d90668a) because it breaks some functionality in Nyx. Honestly, I have no idea what documententation was meant, but we should also consider including the information about the current CPU mode in the IP callback function. This is especially useful if you need to know in which mode your code has been executed...

**Comment by tklengyel** [comment]
What that commit fixed is this how the basic block callback is registered. Notice the cb has 3 arguments:
```
void libxdc_register_bb_callback(libxdc_t* self,  void (*basic_block_callback)(void*, uint64_t, uint64_t), void* basic_block_callback_opaque);
```

This is how you issue the callback, with 4 arguments:

```
		self->basic_block_callback(self->basic_block_callback_opaque, mode, self->cfg.base_addr[res_nid], self->cfg.cofi_addr[res_nid]);
```

**Comment by tklengyel** [comment]
By documentation I meant the public header. Note that nothing in the public header ever uses `disassembler_mode_t`.

**Comment by schumilo** [comment]
Okay, I see…
Seems like the function prototypes in the public headers are simply wrong (`disassembler_mode_t` is missing everywhere). Same for the testing code. Note that we had intentionally defined 4 arguments for the callback function. Otherwise we won't get the information about the currently used CPU mode. 
Here is an example from the Nyx codebase of how we actually use the code: https://github.com/nyx-fuzz/QEMU-Nyx/blob/qemu-nyx-4.2.0/nyx/redqueen.c#L208

So long story short: Thanks for pointing out the error :)
I'll fix the code later today.

**Comment by schumilo** [comment]
Shall we update the other callback functions the same way (add an additional `mode` argument)?

**Comment by tklengyel** [comment]
I don't have a use for `mode` but if its needed in some usecase then updating the public header would be fine.
